webapp/src/components/post_type_transcription.tsx

```diff
@@ -1,23 +1,23 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
 import {useSelector} from 'react-redux';
 
+import type {Post} from 'mattermost-redux/types/posts';
+
 import styled from 'styled-components';
 
 import IconAI from 'src/components/ai_icon';
 
+const aiPluginID = 'mattermost-ai';
+
 const useAIAvailable = () => {
-    //@ts-ignore plugins state is a thing
-    return useSelector((state) => Boolean(state.plugins?.plugins?.[aiPluginID]));
+    return useSelector((state: any) => Boolean(state.plugins?.plugins?.[aiPluginID]));
 };
 
-const aiPluginID = 'mattermost-ai';
-
 const useCallsPostButtonClicked = () => {
-    return useSelector((state) => {
-        //@ts-ignore plugins state is a thing
-        return state['plugins-' + aiPluginID]?.callsPostButtonClickedTranscription;
+    return useSelector((state: any) => {
+        const aiPluginState = state['plugins-' + aiPluginID];
+        return aiPluginState?.callsPostButtonClickedTranscription;
```

Comment:
We want some type info available to the caller

```suggestion
        return aiPluginState?.callsPostButtonClickedTranscription as (() => void) | undefined;
```

Reply:
Done

Reply:
Something in the current versions that we use in the plugin is not allowing me to do it properly, so I just removed that. Sorry, but I think the problem is the current status of the dependencies of the project so I would suggest to handled that through updating the dependencies, and then we can handle this cases.